backtrace-library/src/main/java/backtraceio/library/BacktraceClient.java

```diff
@@ -292,4 +330,18 @@ public void disableAnr() {
             this.anrWatchdog.stopMonitoringAnr();
         }
     }
+
+    public void onPause() {
+        if(monitoringBattery) {
+            if (batteryStateHelper != null)
+                context.unregisterReceiver(batteryStateHelper);
+        }
+    }
+
+    public void onResume() {
+        if(monitoringBattery) {
```

Comment:
Most of the logic in this function should probably be moved to BatteryStateHelper